components/dashboards-web-component/src/viewer/components/DashboardReportGenerationCard.jsx

```diff
@@ -0,0 +1,132 @@
+                    <div style={{ marginRight: 0 }}>
+                        <RaisedButton
+                            label='Capture current page'
+                            onClick={this.capturePage}
+                            primary
+                        />
+
+                        <List>
+                            {this.state.pageList.map(field =>
+                                (<ListItem
+primaryText={
+                                    field}
+                                    rightIcon={<DeleteIcon />}
+                                onClick={() => this.removePage(field)}
+                                />),
+                            )}
+                        </List>
+
+                        <SelectField
+                            style={{ width: 200 }}
+                            floatingLabelText='Page Size'
+                            value={this.state.pageSize}
+                            onChange={(event, index, value) => { this.setState({ pageSize: value }); }}
+
+                        >
+                            {pageSizes.map(field =>
+                                (
+                                    <MenuItem
+                                        key={field}
+                                        value={field}
+                                        primaryText={field}
+                                    />
+                                ))}
+                        </SelectField>
+
+                        <ReportGenerationButton
+                            pageSize={this.state.pageSize}
+                            pageList={this.state.pageList}
+                            pages={this.props.pages}
+                        />
+                    </div>
+                </CardActions>
+            </Card>
+
+        );
+    }
+
+    handleCardClick(expand) {
+        this.setState({ expanded: expand });
+    }
+
+    removePage(index) {
+        const tempPageList = this.state.pageList.slice();
+        const ind = tempPageList.indexOf(index);
+        tempPageList.splice(ind, 1);
+        this.setState({ pageList: tempPageList });
+
+        if (tempPageList.indexOf(index) == -1) {
+            localStorage.removeItem(index);
+        }
+    }
+
+    capturePage(index) {
+        const url = window.location.href;
+        const parts = url.split('/');
+        let currentPage = parts[parts.length - 1] === '' ? parts[parts.length - 2] : parts[parts.length - 1];
+        currentPage = this.props.pages.find((page) => {
+            return page.id === currentPage;
+        });
+
+        html2canvas(document.getElementById('dashboard-container')).then((canvas) => {
+            const imgData = canvas.toDataURL('image/png');
+            const w = canvas.width;
+            const h = canvas.height;
+            localStorage.setItem(currentPage.name, JSON.stringify({ imageData: imgData, width: w, height: h }));
```

Comment:
Don't put the data to local storage with page name as the key. Lets prefix the current page name with some string (namespacing). This is to void conflicts.

e.g. prefix: `_dashboard-report:`
Refer: https://github.com/wso2/carbon-dashboards/blob/master/components/dashboards-web-component/src/utils/DashboardThumbnail.js#L33

Reply:
Done